exe/json2yaml.hs

```diff
@@ -9,8 +8,9 @@ import qualified Data.Yaml as Y
 main :: IO ()
 main = do
     args <- getArgs
-    when (length args > 2) $ error "Usage: json2yaml [in] [out]"
-    let (input:output:_) = args ++ repeat "-"
+    let (input, output) = case args ++ repeat "-" of
```

Comment:
This is definitely better, but a small request for further improvement: I'd rather see:

```haskell
(input, output) <- case ... of
  (i:o:_) -> return (i, o)
  _ -> error ...
```

Since this ensures that the error is thrown immediately. Also: it may be better to pattern match for _exactly_ 2 values `[i, o]`. I realize you're just copying the old behavior :)

Reply:
Sure!

Reply:
Sorry, I was being stupid before (I'll blame it on jet lag). Because of the `++ repeat` approach I used before, this won't work. Instead, the pattern matching would have to be something like:

```
case args of
  [] -> return ("-", "-")
  [i] -> return (i, "-")
  [i, o] -> return (i, o)
  _ -> error "Usage: ..."
```

Reply:
I see